### PR TITLE
Make server core module with pure functions

### PR DESCRIPTION
**Pull request by prabodh1194**
Address #2 

Vote and revote functions now return a set of objects structured same as
object state itself, but actually just carry changes and are merged with
state object in reducer module.

### CODE REVIEW

**Comment by prabodh1194** [comment]
ooohk. No training with redux. I guess I don't understand what exactly does pure function mean in this case.

File: src/server/reducer.js [hunk 1]
```diff
@@ -5,10 +5,42 @@ function reducer(state = {}, action) {
     case 'init_state':
       return core.initState(action.payload);
     case 'vote':
-      return core.vote(state, action.payload, action.clientId);
+      returnSet = core.vote(state, action.payload, action.clientId);
```
Comment (eugrdn) [inline src/server/reducer.js]: We don't need to make any changes here. _Vote_ fn should be enough to creating new state.

**Comment by eugrdn** [comment]
@prabodh1194 
A pure function mean:
- idempotent in the programming sense, since it cannot have any side effects
- the same output for the same input
- referential transparency

Check some articles about FP in JS

**Comment by prabodh1194** [comment]
Ok. I need to read up redux. I guess i'll leave this request someone experienced in redux. Feel free to close.

File: src/server/core.js [hunk 2]
```diff
@@ -9,29 +9,42 @@ function initTopics(topics) {
   }, {});
 }
 
+// return a changeset to merge with state
 function vote(state, topicId, voterId) {
+  returnSet = {'topics': {}, 'voters': []}
   const voter = state.voters.find(voter => voter.id === voterId);
   if (!voter) {
-    state.topics[topicId].votes++;
-    state.voters.push({ id: voterId, vote: topicId });
```
Comment (eugrdn) [inline src/server/core.js]: #push creating a side-effect, also like a directly changing topics object property in line above

**Comment by eugrdn** [comment]
It isn't specifically belongs to redux. Redux use this as a feature, that allow to transparently manage the state of application. Its creator borrowed this from the FP.

**Pull request closed without merging**